src/sdc11073/consumer/components.py

```diff
@@ -53,6 +55,7 @@ def _merge(attr_name: str):
             for handler in other.service_handlers:
                 if handler not in self.service_handlers:
                     self.service_handlers.append(handler)
+        self.additional_schema_specs.extend(other.additional_schema_specs)
```

Comment:
does not check for duplicates. can `additional_schema_specs` be a set? then this would be very easy

Reply:
Implemented merge functionality with sets, but kept type as list for convenience of user.
Merging with builtin schemas is now also implemented with sets (providerimpl and consumerimpl).